src/gpu/intel/jit/gemm/generator/pieces/post_ops.cxx

```diff
@@ -301,28 +299,57 @@ bool BLASKernelGenerator<hw>::gemmBinaryOpC(BinaryOp op, bool row, bool column,
         if (checkRemY)
             cmp(simt | gt | state.flagAP, remY, 0);
 
-        for (int y = 0; y < unrollY; y++) {
-            if (checkRemY) {
-                simtCF ? goto12(16 | ~state.flagAP, lDone)
-                       :   jmpi(1  | ~state.flagAP, lDone);
+        const int grouped_ops = 16;
+        for (int y0 = 0; y0 < unrollY; y0 += grouped_ops) {
+            auto all_CO_regs = std::vector<ngen::GRFRange>(grouped_ops);
+            Label lDoneLoading;
+            // Perform loads
+            for (int i = 0; i < grouped_ops; i++) {
+                const int y = y0 + i;
+                if (y >= unrollY) break;
+                auto &CO_regs = all_CO_regs[i];
+                CO_regs = state.ra.alloc_range(getRegCount(CO_layout));
```

Comment:
Following up on Roy's suggestion, you can pull this allocation out of the `y0` loop, and use `tryAllocRange` instead -- if you run out of registers, just use what you have and limit `grouped_ops` to match.